File: spark-job/src/test/scala/com/dqs/config/RuleConfigTest.scala
package com.dqs.config

import org.scalatest.funsuite.AnyFunSuite

class RuleConfigTest extends AnyFunSuite {
  test("parse rule config json") {
    val json = """{"dataset": "UET0", "checks": [{"checkType": "rowCount", "min": [100]}]}"""
    val config = RuleConfig.fromJson(json)

    assert(config.dataset == "UET0")
    assert(config.checks.size == 1)
    assert(config.checks.head.checkType == "rowCount")
    assert(config.checks.head.min.contains(100))
  }
}

File: spark-job/src/test/java/com/dqs/config/RuleConfigTest.java
package com.dqs.config;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class RuleConfigTest {

    @Test
    void parseRuleConfigJson() {
        String json = "{\"dataset\": \"UET0\", \"checks\": [{\"checkType\": \"rowCount\", \"min\": 100}]}";
        RuleConfig config = RuleConfig.fromJson(json);

        assertEquals("UET0", config.getDataset());
        assertEquals(1, config.getChecks().size());
        assertEquals("rowCount", config.getChecks().get(0).getCheckType());
        assertEquals(100, config.getChecks().get(0).getMin());
    }

    @Test
    void parseRuleConfigWithMinAbsent() {
        String json = "{\"dataset\": \"UET0\", \"checks\": [{\"checkType\": \"rowCount\"}]}";
        RuleConfig config = RuleConfig.fromJson(json);

        assertNull(config.getChecks().get(0).getMin());
    }

    @Test
    void parseRuleConfigMalformedJsonThrows() {
        assertThrows(IllegalArgumentException.class, () -> RuleConfig.fromJson("{bad json}"));
    }
}
